app/src/androidTest/java/com/github/swent/echo/compose/components/EventInfoSheetTest.kt

```diff
@@ -0,0 +1,137 @@
+import com.github.swent.echo.data.model.Location
+import java.util.Date
+import org.junit.Rule
+import org.junit.Test
+
+class EventInfoSheetTest {
+
+    @get:Rule val composeTestRule = createComposeRule()
+
+    private var joinClicked = 0
+    private var peopleClicked = 0
+    private var dismissed = 0
+    private var extended = 0
+
+    private fun setUp() {
+        composeTestRule.setContent {
+            joinClicked = 0
+            peopleClicked = 0
+            dismissed = 0
+
+            val event =
+                Event(
+                    eventId = "1",
+                    organizerId = "1",
+                    title = "Event Title",
+                    description = "Event Description",
+                    location = Location("", 0.0, 0.0),
+                    startDate = Date(2024, 5, 18, 18, 15),
+                    endDate = Date(2024, 1, 1, 22, 0)
+                )
+
+            EventInfoSheet(
+                event = event,
+                hostName = "Event Organization",
+                eventImage = android.R.drawable.ic_menu_help,
+                eventPeople = 0,
+                eventPeopleMax = 0,
+                onJoinButtonPressed = { joinClicked++ },
+                onShowPeopleButtonPressed = { peopleClicked++ },
+                onDismiss = { dismissed++ },
+                onFullyExtended = { extended++ }
+            )
+        }
+    }
+
+    @Test
+    fun shouldShowEventTitle() {
+        setUp()
+        composeTestRule.onNodeWithText("Event Title").assertExists()
+    }
+
+    @Test
+    fun shouldShowEventDescription() {
+        setUp()
+        composeTestRule.onNodeWithText("Event Description").assertExists()
+    }
+
+    @Test
+    fun shouldShowEventDateTime() {
+        setUp()
+        composeTestRule.onNodeWithText("18/06\n" + "18:15").assertExists()
+    }
+
+    @Test
+    fun shouldShowEventOrganization() {
+        setUp()
+        composeTestRule.onNodeWithText("Event Organization").assertExists()
+    }
+
+    @Test
+    fun shouldShowJoinButton() {
+        setUp()
+        composeTestRule.onNodeWithText("Join Event").assertExists()
+    }
+
+    @Test
+    fun shouldShowPeopleButton() {
+        setUp()
+        composeTestRule.onNodeWithText("0").assertExists()
+    }
+
+    @Test
+    fun shouldCallJoinButtonPressedWhenJoinButtonClicked() {
+        setUp()
+        composeTestRule.onNodeWithTag("join_button").performClick()
+        assert(joinClicked == 1)
+    }
+
+    @Test
+    fun shouldCallShowPeopleButtonPressedWhenPeopleButtonClicked() {
+        setUp()
+        composeTestRule.onNodeWithTag("people_button").performClick()
+        assert(peopleClicked == 1)
+    }
+
+    @Test
+    fun shouldShowEventImage() {
+        setUp()
+        composeTestRule.onNodeWithTag("event_image").assertExists()
+    }
+
+    @Test
+    fun shouldShowPeopleCorrectlyWhenMaxPeopleIsNotZero() {
+        // not using setUp() because we need to set eventPeopleMax to 5
```

Comment:
Wouldn't it be possible to add a parameter to setup ?

`private fun setUp(eventPeopleMax: Int = 0) {  ... }` and then call `setUp(5)`